### ISSUE TITLE
Salmon alevin errors out when cell barcode length is > 31 base pairs

### ISSUE THREAD

**Issue by connersk** [COMBINE-lab/salmon#943]
# tl;dr
- `salmon alevin` returns an error if the cell barcode length is > 31 base pairs
- This blocks me from using `salmon alevin` for a novel scRNA-seq chemistry that has a 34 base pair cell barcode.
- Previously the `salmon alevin` maximum cell barcode length was increased from 20 base pairs to 31 base pairs. https://github.com/COMBINE-lab/salmon/discussions/629
- Could you please increase the maximum cell barcode length again?
     - Or provide guidance on how I should do this in my own fork of salmon?

# Summary

This bug primarily related to alevin (single-cell mode).

**Describe the bug**

**To Reproduce**
```
salmon alevin
    -I /path/to/salmon_index 
    -p 16 
     -l IU  
    --read-geometry 2[1-100]        
    --bc-geometry 1[1-34]         
    --umi-geometry 1[35-44]         
    --sketch         
    -1 /path/to/r1.fastq 
    -2 /path/to/r2.fastqs
    -o /output/path        
     --tgMap /path/to/t2g.tsv
```

Steps and data to reproduce the behavior:

Specifically, please provide at least the following information:

  * Salmon version`v1.10.1`
  * How was salmon installed? -> Compiled
  * Encountered this error when running a feature barcoding analysis following this tutorial https://combine-lab.github.io/alevin-fry-tutorials/2021/af-feature-bc/

**Expected behavior**
Salmon alevin does not error out when the cell barcode length is 34 base pairs.

**Screenshots**
Error log
`[2024-06-17 22:00:25.466] [alevinLog] [error] Barcode length (34) was not in the required length range [1, 31].`

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04

**Comment by connersk** [comment]
Following up on this, if I was to fork `salmon` and fix this, would the correct solution be to update the following line? https://github.com/COMBINE-lab/salmon/blob/a2f6912b3f9f9af91e3a4b0d74adcb3bdc4c9a32/src/AlevinUtils.cpp#L1204
